tools/site-query/scripts.js

```diff
@@ -319,15 +319,13 @@ async function init(doc) {
       for await (const sitemapUrl of sitemapUrls) {
```

Comment:
This could all be much faster if you put all the promises in an array and just awaited them.
```
const processingTasks = [];
for await (const sitemapUrl of sitemapUrls) {
  if (sitemapUrl.pathname.startsWith(path)) {
    searched += 1;
    const promise = processUrl(sitemapUrl, query, queryType, org, site).then((tr) => {
      if (tr) {
        results.append(tr);
        updateCaption(caption, results.children.length, searched);
      }
    });
    processingTasks.push(promise);
  }
}
await Promise.all(processingTasks);
```

Reply:
Thanks for the feedback. Code updated.